ats/atsMachines/lsf_asq.py

```diff
@@ -285,11 +291,14 @@ def set_nt_cpus_per_task_num_nodes(self,test):
         # multiple nodes (hosts)
         if ((test.np * test.nt) > self.npMax):
             if test.num_nodes < 1:
-                # if (self.jsrun_nn < 1):
-                test.num_nodes = math.ceil( (float(test.np) * float(test.nt)) / float(self.npMax))
-                test.nn = test.num_nodes
-                if configuration.options.verbose:
-                    print("ATS setting test.nn to %i for test %s based on test.np = %i and test.nt=%i (%i x %i = %i) which spans 2 or more nodes." %
+                if self.lrun_pack and self.lrun_pack == True:
```

Comment:
I am confused by `if self.lrun_pack and self.lrun_pack == True` as it seems redundant.
If `self.lrun_pack` must be `True` and nothing else it would be sufficient to say either...

`self.lrun_pack == True`
or
`self.lrun_pack is True`

Reply:
It seems redudant  but is not.  lrun_pack  the firest checks to see if it actually exists, as it may not exist.  Then if it does exist, is it True.

Reply:
The problem is `if self.lrun_pack` is the same as saying `if bool(self.lrun)` and will evaluate true/false in a lot of cases that may not be correct here.  It is not just checking for if it exists.

For example:

```
value = "some string"
if value:
   print("inside if") // will print
```

You should do the following:

```
if self.lrun_pack is not None and self.lrun_pack == True:
```

Note: I make this w/o looking into what lrun_pack actually is and this is your code, i just got an email =)